runtime/src/main/java/com/datastax/oss/quarkus/runtime/internal/quarkus/CassandraClientProducer.java

```diff
@@ -51,7 +51,15 @@
   @Produces
   @ApplicationScoped
   @Unremovable
-  public QuarkusCqlSession createCassandraClient() {
+  public QuarkusCqlSessionState quarkusCqlSessionState() {
+    return QuarkusCqlSessionState.notInitialized();
+  }
+
+  @Produces
+  @ApplicationScoped
+  @Unremovable
+  public QuarkusCqlSession createCassandraClient(QuarkusCqlSessionState quarkusCqlSessionState) {
+    quarkusCqlSessionState.setInitialized();
```

Comment:
It would probably be more correct if this was the last statement in this method, right before `return session;`.